app/legacy/v2/notifications/utils.py

```diff
@@ -54,14 +71,130 @@ async def validate_template(template_id: str) -> Template:
     """
     # call dao to get template
     # return the template if it exists for given service, else raise an exception
-    raise NotImplementedError('validate_template has not been implemented.')
+    raise NotImplementedError('validate_push_template has not been implemented.')
 
 
-async def get_arn_from_icn(icn: str) -> str:
-    """Future method to get ARN from ICN from VAProfile.
+async def validate_template(
+    template_id: UUID4,
+    expected_type: NotificationType,
+    personalisation: dict[str, str | int | float | list[str | int | float] | PersonalisationFileObject] | None,
+) -> None:
+    """Validates the template with the given ID.
+
+    Checks for the template in the database, that it's the right type, and is active. Raises an HTTPException if any
+    of these conditions are not met.
 
     Args:
-        icn (str): Recipient's ICN Value
+        template_id (UUID4): The template_id to validate
+        expected_type (NotificationType): The expected type of the template
+        personalisation (dict[str, str | int | float | list[str | int | float] | PersonalisationFileObject] | None):
+            The personalisation data to validate
+
+    Raises:
+        ValueError: If the template is not found, or is invalid based on the expected type,
+            personalisation is missing, or is archived.
 
     """
-    raise NotImplementedError('get_arn_from_icn has not been implemented.')
+    try:
+        template = await LegacyTemplateDao.get_template(template_id)
+    except NoResultFound:
+        logger.exception('Template not found with ID {}', template_id)
+        raise ValueError('Template not found')
+
+    try:
+        _validate_template_type(template.template_type, expected_type, template_id)
```

Comment:
I guess same question as [above](https://github.com/department-of-veterans-affairs/va-enp-api/pull/195/files#r1991931655). This could return True/False - but I feel it would make the code more verbose